modules/library/render/src/main/java/org/geotools/renderer/lite/StreamingRenderer.java

```diff
@@ -834,6 +830,14 @@ public void paint(
                         "Cannot call paint, you did not set a MapContent in this renderer");
             }
 
+            // handle the background color specification, if any
+            Style firstStyle = mapContent.layers().get(0).getStyle();
+            if (firstStyle.getBackground() != null) {
+                Paint background = styleFactory.getPaint(firstStyle.getBackground(), null, null);
+                graphics.setPaint(background);
+                graphics.fill(paintArea);
```

Comment:
Do we need to make the image transparent, before applying your fill, in case the fill has some transparency effects?

Reply:
No, StreamingRenderer is given a Graphic2D, it should not alter what was done before on it (could have been used for overlay on top of other stuff for example). It should be simple and predictable, just do what it was asked to do (apply the background).

Reply:
thanks that makes sense